File: binary_search.go
func search(nums []int, target int) int {
    if (len(nums) == 1) {
        if (nums[0] == target) {
            return 0;
        }
        return -1;
    }
    left := 0;
    right := len(nums) - 1;
    for (true) {
        if (left + 1 == right) {
            if (nums[left] == target) {
                return left;
            }
            if (nums[right] == target) {
                return right;
            }
            break;
        }
        mid := (left + right) / 2;
        if (nums[mid] == target) {
            return mid;
        } else if (nums[mid] < target) {
            left = mid;
        } else {
            right = mid;
        }
    }
    return -1;
}

File: Algorithm_I_study_plan/binary_search.py
from typing import List

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1
        while l < r:
            mid = (l + r) // 2
            if nums[mid] == target:
                return mid
            if nums[mid] < target:
                l = mid + 1
            if nums[mid] > target:
                r = mid - 1
        if r == l and nums[r] == target:
            return r
        return -1